lib/ansible/modules/network/fortios/fortios_address_group.py

```diff
@@ -0,0 +1,236 @@
+  fortios_address_group:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: present
+    name: dns_servers
+    member: google_dns
+    comment: "Public DNS servers"
+
+- name: Remove yahoo from trusted mail providers
+  fortios_address_group:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: absent
+    name: trusted_mail_providers
+    member: yahoo_server
+    comment: "Trusted Email providers"
+"""
+
+RETURN = """
+firewall_address_config:
+  description: full firewall adresses config string
+  returned: always
+  type: string
+change_string:
+  description: The commands executed by the module
+  returned: only if config changed
+  type: string
+"""
+
+import re
+import shlex
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+
+from ansible.module_utils.fortios import fortios_argument_spec, fortios_required_if
+from ansible.module_utils.fortios import backup
+
+#check for pyFG lib
+try:
+    from pyFG import FortiOS, FortiConfig
+    from pyFG.exceptions import CommandExecutionException, FailedCommit
+    HAS_PYFG=True
+except:
+    HAS_PYFG=False
+
+
+def is_invalid_name(input_str):
+    #char must be letters, digits, - , _ , .
+    #must have between 1 and 63 chars
+    reg=re.compile(r'^[a-zA-Z0-9\.\-_]{1,63}$')
+    return not reg.match(input_str)
+
+def main():
+    #define module params
+    argument_spec = dict(
+        state       = dict(choices=['present', 'absent'], default='present'),
+        name        = dict(aliases=['group_name'], required=True, type='str'),
+        member      = dict(aliases=['members'], required=True, type='list'),
+        comment     = dict(type='str'),
+    )
+
+    #merge global argument_spec from module_utils/fortios.py
+    argument_spec.update(fortios_argument_spec)
+
+    #decalre module
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=fortios_required_if,
+    )
+
+    #check params
+    if is_invalid_name(module.params['name']):
+        module.fail_json(msg="Bad name argument value %s, must contain only letters, digits, -, _, ." % (module.params['name']))
+
+    for member in module.params['member']:
+        if is_invalid_name(member):
+            module.fail_json(msg="Bad member argument value %s, must contain only letters, digits, -, _, ." % (member))
+
+    #prepare return dict
+    result = dict(changed=False)
+
+    # fail if PYFG not present
+    if not HAS_PYFG:
+        module.fail_json(msg='Could not import the python library pyFG required by this module')
+
+    #define device
+    f = FortiOS( module.params['host'],
+        username=module.params['username'],
+        password=module.params['password'],
+        timeout=module.params['username'],
+        vdom=module.params['vdom'])
+
+    path = 'firewall addrgrp'
+
+    #connect
```

Comment:
There is some code duplicated between this pull-request and other FortisOS pull-requests ([`fortios_ipv4_policy`](https://github.com/ansible/ansible/pull/21849/files/d49860f735c162acda87f5232f1de0e148453203#r103181410) and [`fortios_address`](#21542)):

- block calling `connect` method
- block calling `load_config` method

These blocks must be moved together in [`module_utils/fortios.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/fortios.py).